x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/components/all_exception_items_table/index.tsx

```diff
@@ -0,0 +1,393 @@
+    pageIndex: 0,
+    pageSize: 25,
+    totalItemCount: 0,
+    pageSizeOptions: [1, 5, 10, 25, 50, 100, 200, 300],
+  },
+  exceptions: [],
+  exceptionToEdit: null,
+  currenFlyout: null,
+  viewerState: 'loading',
+};
+
+export interface GetExceptionItemProps {
+  pagination?: Partial<Pagination>;
+  search?: string;
+  filters?: string;
+}
+
+interface ExceptionsViewerProps {
+  rule: Rule | null;
+  listType: ExceptionListTypeEnum;
+  onRuleChange?: () => void;
+}
+
+const ExceptionsViewerComponent = ({
+  rule,
+  listType,
+  onRuleChange,
+}: ExceptionsViewerProps): JSX.Element => {
+  const { services } = useKibana();
+  const toasts = useToasts();
+  const [{ canUserCRUD, hasIndexWrite }] = useUserData();
+  const [isReadOnly, setReadOnly] = useState(true);
+  const [lastUpdated, setLastUpdated] = useState<null | string | number>(null);
+  const exceptionListsToQuery = useMemo(
+    () =>
+      rule != null && rule.exceptions_list != null
+        ? rule.exceptions_list.filter((list) => list.type === listType)
+        : [],
+    [listType, rule]
+  );
+
+  // Reducer state
+  const [{ exceptions, pagination, currenFlyout, exceptionToEdit, viewerState }, dispatch] =
+    useReducer(allExceptionItemsReducer(), {
+      ...initialState,
+    });
+
+  // Reducer actions
+  const setExceptions = useCallback(
+    ({
+      exceptions: newExceptions,
+      pagination: newPagination,
+    }: UseExceptionListItemsSuccess): void => {
+      setLastUpdated(Date.now());
+
+      dispatch({
+        type: 'setExceptions',
+        exceptions: newExceptions,
+        pagination: newPagination,
+      });
+    },
+    [dispatch]
+  );
+
+  const setViewerState = useCallback(
+    (state: ViewerState): void => {
+      dispatch({
+        type: 'setViewerState',
+        state,
+      });
+    },
+    [dispatch]
+  );
+
+  const setFlyoutType = useCallback(
+    (flyoutType: ViewerFlyoutName): void => {
+      dispatch({
+        type: 'updateFlyoutOpen',
+        flyoutType,
+      });
+    },
+    [dispatch]
+  );
+
+  const [_, allReferences] = useFindExceptionListReferences(exceptionListsToQuery);
+
+  const handleFetchItems = useCallback(
+    async (options?: GetExceptionItemProps) => {
+      const abortCtrl = new AbortController();
+
+      const newPagination =
+        options?.pagination != null
+          ? {
+              page: (options.pagination.page ?? 0) + 1,
+              perPage: options.pagination.perPage,
+            }
+          : {
+              page: pagination.pageIndex + 1,
+              perPage: pagination.pageSize,
+            };
+
+      if (exceptionListsToQuery.length === 0) {
+        return {
+          data: [],
+          pageIndex: pagination.pageIndex,
+          itemsPerPage: pagination.pageSize,
+          total: 0,
+        };
+      }
+
+      const {
+        page: pageIndex,
+        per_page: itemsPerPage,
+        total,
+        data,
+      } = await fetchExceptionListsItemsByListIds({
```

Comment:
is any error handling needed here?

Reply:
This is used to wrap around - so where it gets invoked is where there's the try/catch. @WafaaNasr is also working on moving out this logic so it may differ with her updates.